src/persist/src/indexed/columnar/arrow.rs

```diff
@@ -117,6 +99,59 @@ pub(crate) const ENABLE_ARROW_LGALLOC_NONCC_SIZES: Config<bool> = Config::new(
     "A feature flag to enable copying decoded arrow data into lgalloc on non-cc sized clusters.",
 );
 
+fn realloc_data(data: ArrayData, metrics: &ColumnarMetrics) -> ArrayData {
+    // NB: Arrow generally aligns buffers very coarsely: see arrow::alloc::ALIGNMENT.
+    // However, lgalloc aligns buffers even more coarsely - to the page boundary -
+    // so we never expect alignment issues in practice. If that changes, build()
+    // will return an error below, as it does for all invalid data.
```

Comment:
This feels a little sketchy to me, though it also seems fine in practice. Please let me know if it seems better to tackle this differently however!

Reply:
Works for me!

Reply:
lgalloc guarantees page-alignment, but it cannot much else as it's bound by whatever mmap choses to do, which doesn't have an alignment parameter.